tensorflow/python/keras/mixed_precision/experimental/autocast_variable_test.py

```diff
@@ -318,18 +311,19 @@ def run_and_check():
         self.assertAllClose(3., self.evaluate(x.assign_sub(3.)))
 
         # Assign multiple times
-        assign = x.assign(1.)
-        self.assertAllClose(1., self.evaluate(assign))
-        self.assertAllClose(0., self.evaluate(assign.assign(0.)))
-        assign_add = x.assign_add(3.)
-        self.assertAllClose(3., self.evaluate(assign_add))
-        self.assertAllClose(3. * 3,
-                            self.evaluate(x.assign_add(3.).assign_add(3.)))
-        self.assertAllClose(3. * 3, x)
-        assign_sub = x.assign_sub(3.)
-        self.assertAllClose(3. * 2, self.evaluate(assign_sub))
-        self.assertAllClose(0.,
-                            self.evaluate(x.assign_sub(3.).assign_sub(3.)))
+        if not ds_context.has_strategy():
+          assign = x.assign(1.)
+          self.assertAllClose(1., self.evaluate(assign))
+          self.assertAllClose(0., self.evaluate(assign.assign(0.)))
+          assign_add = x.assign_add(3.)
+          self.assertAllClose(3., self.evaluate(assign_add))
+          self.assertAllClose(3. * 3,
+                              self.evaluate(x.assign_add(3.).assign_add(3.)))
+          self.assertAllClose(3. * 3, x)
+          assign_sub = x.assign_sub(3.)
+          self.assertAllClose(3. * 2, self.evaluate(assign_sub))
+          self.assertAllClose(0.,
+                              self.evaluate(x.assign_sub(3.).assign_sub(3.)))
```

Comment:
Now that the distributions strategy used for the test has two devices this test correctly fails with:
```python-traceback
/keras/mixed_precision/experimental/autocast_variable_test.runfiles/org_tensorflow/tensorflow/python/distribute/values.py", line 197, in __getattr__
    return getattr(self._get(), name)
AttributeError: 'Tensor' object has no attribute 'assign'
```

@reedwm I am happy to look into a fix for this in a future PR. I think it should be possible to fix for the eager and tf_function case without too much changes to `AutCastVariable`.

Reply:
AFAIK, this can only be fixed once `DistributionVariable.assign` returns a variable. If you have a way to fix this before then, let me know. I don't want to make AutoCastVariable more complicated than necessary since I think the distrubtion strategy team will fix the `DistributionVariable.assign` issue fairly soon, but I'm ok fixing it before then if the change would be fairly simple.

Reply:
Makes sense. In that case my comment might have been a bit to confident, since I haven't looked at it in detail yet. I will have a quick look at it tomorrow and let you know then

Reply:
@reedwm It looks like a quick fix is possible. Checkout https://github.com/lgeiger/tensorflow/commit/5b5ab0034ae0f12731943177b275e35cb9f92bb1 which seems to work well and probably also slightly reduces the overhead of creating a new `AutoCastVariable` instance on every variable assignment. If it looks reasonable to you, I will go ahead and open a separate PR once this one is merged.